lib/routes.js

```diff
@@ -135,14 +135,27 @@ module.exports = async function (app, opts) {
   const errorFormatter = typeof opts.errorFormatter === 'function' ? opts.errorFormatter : defaultErrorFormatter
 
   if (typeof opts.errorHandler === 'function') {
-    app.setErrorHandler(opts.errorHandler)
+    app.setErrorHandler((error, request, reply) => {
+      const errorHandler = opts.errorHandler
+      if (!request[kRequestContext]) {
+        // Generate the context for this request
+        request[kRequestContext] = { reply, app }
```

Comment:
sorry for missing this.

@mcollina 
should not we try to generate full context, like in other places?

```ts
    if (contextFn) {
      request[kRequestContext] = await contextFn(request, reply)
      Object.assign(request[kRequestContext], { reply, app })
    } else {
      request[kRequestContext] = { reply, app }
    }
```

Reply:
I just focused on fixing it asap. Open a fresh PR in case.